pkg/storage/chunk/azure/blob_storage_client.go

```diff
@@ -234,39 +237,97 @@ func (b *BlobStorage) buildContainerURL() (azblob.ContainerURL, error) {
 }
 
 func (b *BlobStorage) newPipeline(hedging bool) (pipeline.Pipeline, error) {
-	credential, err := azblob.NewSharedKeyCredential(b.cfg.AccountName, b.cfg.AccountKey.Value)
+	if b.cfg.UseManagedIdentity == false {
+		credential, err := azblob.NewSharedKeyCredential(b.cfg.AccountName, b.cfg.AccountKey.Value)
+		if err != nil {
+			return nil, err
+		}
+
+		opts := azblob.PipelineOptions{
+			Retry: azblob.RetryOptions{
+				Policy:        azblob.RetryPolicyExponential,
+				MaxTries:      (int32)(b.cfg.MaxRetries),
+				TryTimeout:    b.cfg.RequestTimeout,
+				RetryDelay:    b.cfg.MinRetryDelay,
+				MaxRetryDelay: b.cfg.MaxRetryDelay,
+			},
+		}
+
+		if hedging {
+			opts.HTTPSender = pipeline.FactoryFunc(func(next pipeline.Policy, po *pipeline.PolicyOptions) pipeline.PolicyFunc {
+				client := b.hedgingCfg.Client(defaultClient)
+				return func(ctx context.Context, request pipeline.Request) (pipeline.Response, error) {
+					resp, err := client.Do(request.WithContext(ctx))
+					return pipeline.NewHTTPResponse(resp), err
+				}
+			})
+		}
+
+		return azblob.NewPipeline(credential, opts), nil
+	} else {
```

Comment:
nit: since the if clause ends with a return, you can safely move the else code to outside of it.
```suggestion
	}
```